.github/workflows/ci.yml

```diff
@@ -508,10 +510,29 @@ jobs:
         exclude_file: .codespell.exclude
 
 
-
   style-check:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
     - uses: editorconfig-checker/action-editorconfig-checker@main
     - run: editorconfig-checker
+
+
+  check-coding-style:
+    needs: changed
+    if: ${{ needs.changed.outputs.game_code == 'true' || needs.changed.outputs.unit_tests == 'true' }}
+    runs-on: ubuntu-latest
+    steps:
+    - name: checkout
+      uses: actions/checkout@v2
+    - name: set up python environment
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.10.6'
+    - name: install python packages
+      run: |
+        python -m pip install --upgrade pip
```

Comment:
I'm not sure we need to upgrade pip here? I'm not sure though.

Reply:
I don't know, most of this config is just copy-pasted from a dozen different places. It seems to work, and doesn't take too long, so I don't see what we would gain by removing it.